### ISSUE TITLE
Relative external problem on ES format

### ISSUE THREAD

**Issue by hiendv** [rollup/rollup#1653]
I have an entry file written in ES6 which contains nothing but reexports. You could imagine it like [this](https://github.com/lodash/lodash/blob/b1ff59b3ea86aefb67676e4e3f5ff636c23fd6c7/lodash.js). When I exclude a relative module, the bundle in ES format will generate an export with an absolute path and an unused import with a relative path.
Entry
```js
export { default as bar } from './bar.js'
```

Output
```js
import bar_js from './bar.js';



export { bar } from '/tmp/rollup/bar.js';
```

Expected
```js
export { default as bar } from './bar.js';
```
Repro: https://github.com/hiendv/rollup-repro-relative-externals

**Comment by amenadiel** [comment]
I've seen that behavior too. When you `export` from a relative path, and said path is excluded, e.g

```js
export {something} from './foo.js'
```
the exclusion is resolved prior to the module output, so you will end with an absolute path in the build. In your example, **even if `bar.js` exported a named `{bar}` object,** you'll still end up with

```js
export { bar } from '/tmp/rollup/bar.js';
```
You can't set an exclusion using a relative path. Setting

```js
external:['./foo.js']
```
will result in an error
```sh
[!] Error: Could not resolve './bar.js' from main.js
```

**However** the workaround I've used for these cases is to perform an `import` and `export`
```js
import {
	default as bar
} from './bar.js';

export {
	bar
};
```

your `rollup.config.js` would generate something as 

```js
import bar_js from './bar.js';
export { bar_js as bar };
```
Which is a valid module and should serve your purposes.

You could also use

```js
export {
	default
}
from './bar.js';
```

the resulting module will be

```js
import bar_js from './bar.js';
export default bar_js;
```
still valid, albeit not what you wanted.

**Comment by hiendv** [comment]
@amenadiel Thank you for the workaround! It worked but I think this issue should be addressed and solved.

**Comment by hiendv** [comment]
@amenadiel Well, not actually. I need another workaround for this case.
Entry
```js
export * from './modules.js'
```

Modules.js (The old entry)
```js
import { default as bar } from './bar.js'
export { bar }
```

The result remains the same.

**Comment by guybedford** [comment]
This seems to be resolved in 0.55.4 - https://rollupjs.org/repl?version=0.55.4&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMColMjBmcm9tJTIwJy4lMkZtb2R1bGVzLmpzJyUyMiU3RCUyQyU3QiUyMm5hbWUlMjIlM0ElMjJtb2R1bGVzLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMCU3QiUyMGRlZmF1bHQlMjBhcyUyMGJhciUyMCU3RCUyMGZyb20lMjAnYmFyLmpzJyU1Q25leHBvcnQlMjAlN0IlMjBiYXIlMjAlN0QlMjIlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyZXMlMjIlMkMlMjJuYW1lJTIyJTNBJTIyJTIyJTJDJTIyZ2xvYmFscyUyMiUzQSU3QiU3RCUyQyUyMmFtZCUyMiUzQSU3QiUyMmlkJTIyJTNBJTIyJTIyJTdEJTdEJTJDJTIyZXhhbXBsZSUyMiUzQW51bGwlN0Q=
